src/main/java/teammates/sqllogic/core/UsersLogic.java

```diff
@@ -726,6 +730,8 @@ private boolean isSectionChanged(Section originalSection, Section newSection) {
     /**
      * Updates a student by {@link Student}.
      *
+     * <p>If email changed, update by recreating the student and cascade update all responses
```

Comment:
Can i check why response and comments have to be updated? From what I can tell there isn't a relation between email and these fields

Reply:
I could be mistaken, but I believe user emails are currently used for the giver/receiver fields in the responses, and the giver/lastEditorEmail in the comments, as these entities have not been directly linked to User yet.

Reply:
I'm looking at the code and you seem to be right, though I think this is different from the ER diagram which has a FK relation to User. Let me check with the others and get back to you

Reply:
Ok so the ER diagram is what the relationships would be at the end of the migration, for now we'll be porting the datastore relationships directly and changing it later, so your solution is correct